### DISCUSSION TITLE
Lesson 9 : Fund Subscription

### DISCUSSION THREAD

**Discussion by karthikeh02** [Cyfrin/foundry-full-course-cu#2115] [Q&A]
in terminal ,

**forge script script/Interactions.s.sol:FundSubscription --rpc-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY --broadcast** 

i got this error,

**[⠆] Compiling...
No files changed, compilation skipped
Script ran successfully.

== Logs ==
  Funding Subscription : 13496312956504562871155198840805877279739069202521477031477111625079572782821
  Using vrfCoordinator : 0x9DdfaCa8183c41ad55329BdeeD9F6A8d53168B1B
  On ChainId : 11155111

## Setting up 1 EVM.

==========================

Chain 11155111

Estimated gas price: 69.517015229 gwei

Estimated total gas used for script: 5353305

Estimated amount required: 0.372145785210481845 ETH

==========================

Transactions saved to: /home/karthikeh/Desktop/foundry-smartcontract-lottery-karthik/broadcast/Interactions.s.sol/11155111/run-latest.json

Sensitive values saved to: /home/karthikeh/Desktop/foundry-smartcontract-lottery-karthik/cache/Interactions.s.sol/11155111/run-latest.json

Error: 
Failed to send transaction

Context:
- server returned an error response: error code -32000: future transaction tries to replace pending**

i had 0.21 something sepolia eth in my wallet. is that not enough when deploying this code?
this is my wallet address pic!

![transactions](https://github.com/user-attachments/assets/87886c15-4546-42a8-82bf-bcd484c5dd8b)

see the out for the past 1 day(those 4 continous out) can anyone explain these?
those two contract creation takes up pretty much - 0.18 sepolia eth why these much?
even though the error is occured, the fund subscription for depositing link in chainlink worked!

**Comment by EngrPips** [comment]
Hello @ karthikeh02, I heard that Sepolia testnet has been slow lately, and it seems like it has been costing more to process transactions due to congestion. I don't know what this error `server returned an error response: error code -32000: future transaction tries to replace pending` meant. I also tried to look it up on Google but found nothing meaningful. You should check the Foundry official repo `https://github.com/foundry-rs/foundry ` to see if someone has mentioned this issue, and if not, you can create an issue so the Foundry team can give better clarification.

**Reply by karthikeh02** [reply]
i thought that my wallet balance is insufficient of sepolia ETH , so again i tried after sufficient balance actually the transaction is successful.

**[⠆] Compiling...
No files changed, compilation skipped
Script ran successfully.

== Logs ==
  Funding Subscription : 13496312956504562871155198840805877279739069202521477031477111625079572782821
  Using vrfCoordinator : 0x9DdfaCa8183c41ad55329BdeeD9F6A8d53168B1B
  On ChainId : 11155111

## Setting up 1 EVM.

==========================

Chain 11155111

Estimated gas price: 3.213611356 gwei

Estimated total gas used for script: 5353305

Estimated amount required: 0.01720344174013158 ETH

==========================

##### sepolia
✅  [Success]Hash: 0xf15acb8038574377f46629bca84ea35608c80474e76adf2e89fd31cf4781dc3d
Contract Address: 0xa94aA08742df673Fc30241A0D9013F191b30719e
Block: 6377254
Paid: 0.002024804660313654 ETH (906786 gas * 2.232946539 gwei)


##### sepolia
✅  [Success]Hash: 0x16f7355c2710aecb83134caf3543a9649bd2e102959c3c72dd2c15378bde6c29
Block: 6377254
Paid: 0.000137703580113591 ETH (61669 gas * 2.232946539 gwei)


##### sepolia
✅  [Success]Hash: 0x2457d08045d9f3beaa5f02e94b2d8bccef262fa07ca72cf22127dcd76f90fb96
Contract Address: 0x7DCBFCfa942C78FbB87d1c23A5e29E1ba243ab82
Block: 6377254
Paid: 0.007026705390267909 ETH (3146831 gas * 2.232946539 gwei)

✅ Sequence #1 on sepolia | Total Paid: 0.009189213630695154 ETH (4115286 gas * avg 2.232946539 gwei)
                                                                                                                                   

==========================

ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.

Transactions saved to: /home/karthikeh/Desktop/foundry-smartcontract-lottery-karthik/broadcast/Interactions.s.sol/11155111/run-latest.json

Sensitive values saved to: /home/karthikeh/Desktop/foundry-smartcontract-lottery-karthik/cache/Interactions.s.sol/11155111/run-latest.json**


you can see the last 3 transaction in my wallet regarding this. my question here is why is there a 3 transaction gone through?
and why the transaction fees is less than 0.009 sepolia ETH ? before it was 0.18 ETH?

**Reply by EngrPips** [reply]
The same transaction can cost more sometimes than it will cost some other time. How congested the blockchain is when you process your transaction is what will determine how high the fee you will pay to process your transaction. About having `3` transactions going through for a single `FundSubscription` share the link to see the transactions, But I will assume the transactions are the process to successfully fund the `Subscription`

**Reply by karthikeh02** [reply]
here is the link to see the transactions

[https://sepolia.etherscan.io/address/0x329830b49b41b1f348420eecf79e1a73d8b45fb9](url)

**Reply by EngrPips** [reply]
Ok
